### PR TITLE
197 set up versioneer and test releasing

### LINKED ISSUE #197: Release workflow
MONAI needs to follow good practice for its versioning system and release workflow.

Versions should be numbered following [Semantic Versioning (semver)](https://semver.org/spec/v2.0.0.html). [Versioneer](https://github.com/warner/python-versioneer) can help for this. See e.g. how it is used in [NiftyNet](https://github.com/NifTK/NiftyNet)

The package should be published on [pypi](https://pypi.org/).

Documentation for project admins should bee provided. Niftynet has some [example](https://niftynet.readthedocs.io/en/dev/contributing.html#niftynet-admin-tasks) that can use as starting point.

### PR DESCRIPTION
**Pull request by wyli**
Fixes #197.

### Description
- set up versioneer for monai, 
so that `python -c 'import monai; print(monai.__version__)'`  returns a PEP440 versioning
- merged several codebase configs into `setup.cfg`
- adds a section of the release procedure in `contributing.md`

### Testing
To test the correctness, I've tagged a commit of this branch with version number `0.1a1.dev1`:
- running `python -c 'import monai; print(monai.__version__)'`
   returned the correct version number. this was tested from multiple cases:
   - pip installed package
   - source code
   - asset https://github.com/Project-MONAI/MONAI/archive/0.1a1.dev1.zip
- adding new commits on top of the tag, 
the version number becomes `0.1a1.dev1+1.ge6de7ac`. Which is expected. the number indicates 1 revision ahead of `0.1a1.dev1` with hash e6de7ac.

### Things not covered
the `CONTRIBUTING.md` now covers the building/releasing procedure,
automating the procedure will need another iteration

### Status
**READY**

### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New tests added to cover the changes
- [x] Docstrings/Documentation updated

### CODE REVIEW

File: codecov.yml [hunk 1]
```diff
@@ -26,3 +26,7 @@ coverage:
 
 # Disable comments on PR
 comment: false
+
+ignore:
+  - "versioneer.py"
+  - "monai/_version.py"
```
Comment (Nic-Ma) [inline codecov.yml]: missing a blank line.

File: monai/_version.py [hunk 2]
```diff
@@ -0,0 +1,520 @@
+
```
Comment (Nic-Ma) [inline monai/_version.py]: Should we add MONAI copyright in this file?

File: versioneer.py [hunk 3]
```diff
@@ -0,0 +1,1822 @@
+
```
Comment (Nic-Ma) [inline versioneer.py]: Should we add some statement or explanation here for the versioneer code?
And also the copyright of MONAI.

File: setup.py [hunk 4]
```diff
@@ -1,55 +1,9 @@
-import os
-import io
-import re
-from setuptools import setup, find_packages
+from setuptools import find_packages, setup
```
Comment (Nic-Ma) [inline setup.py]: Should we add copyright for all the python code in MONAI?

File: codecov.yml (hunk 1 above)
Comment (wyli) [reply codecov.yml]: thanks, this now fixed

File: monai/_version.py (hunk 2 above)
Comment (wyli) [reply monai/_version.py]: i don't think so, this file is in public domain we can't claim copyright

File: versioneer.py (hunk 3 above)
Comment (wyli) [reply versioneer.py]: this file is auto generated by versioneer and I feel it's already well documented

File: setup.py (hunk 4 above)
Comment (wyli) [reply setup.py]: thanks now fixed

**Pull request merged**